chalicelib/matrix_handler.py

```diff
@@ -0,0 +1,141 @@
+import json
+import os
+import tempfile
+import boto3
+import hca
+import loompy
+
+from abc import ABC, abstractmethod
+from botocore.exceptions import ClientError
+from chalicelib.constants import MERGED_MTX_BUCKET_NAME, MERGED_REQUEST_STATUS_BUCKET_NAME, JSON_EXTENSION
+from chalicelib.request_handler import RequestHandler, RequestStatus
+from chalicelib import logger
+
+
+class MatrixHandler(ABC):
+    """
+    A generic matrix handler for matrices concatenation
+    """
+    def __init__(self, extension):
+        self._extension = extension
+        self.hca_client = hca.dss.DSSClient()
+
+    def _filter_mtx(self, bundle_uuids):
+        """
+        Filter only matrix files from a list of DSS bundles.
+        :param bundle_uuids: A list of bundle uuids.
+        :return: A list of matrices uuids.
+        """
+        mtx_uuids = []
+
+        # Iterate uuids to query DSS for bundle manifest of each
+        for uuid in bundle_uuids:
+            bundle_manifest = self.hca_client.get_bundle(replica="aws", uuid=uuid)
+
+            # Gather up uuids of all the matrix files we are going to merge
+            for file in bundle_manifest["bundle"]["files"]:
```

Comment:
Consider calling this `file_` or something that's not a keyword.

Reply:
Fixed.